[Bug] AArch64/NEON build fails with GCC 12.2: invalid construction of xsimd::batch_bool from xsimd::batch<int8_t> in SIMDComparisonUtil.h

### Component Selection

- [x] Core Engine (Expression eval, Memory, Vector)
- [ ] Connectors / File Formats (Hive, Parquet, etc.)
- [ ] API / Bindings (Python, etc.)
- [ ] Build
- [ ] Other

### Describe the Bug

Bolt build failed in arm build with gcc 12.

```
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:223:11: note:   no known conversion for argument 1 from ‘xsimd::batch<signed char>’ to ‘xsimd::batch_bool<signed char, xsimd::neon64>&&’
/workspaces/bolt/bolt/functions/lib/SIMDComparisonUtil.h:75:74: error: no matching function for call to ‘xsimd::batch_bool<signed char, xsimd::neon64>::batch_bool(xsimd::batch<signed char>)’
   75 |         xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 48)));
      |                                                                          ^
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:243:9: note: candidate: ‘template<class Tp> xsimd::batch_bool<T, A>::batch_bool(const Tp*) [with T = signed char; A = xsimd::neon64]’ (deleted)
  243 |         batch_bool(Tp const*) = delete;
      |         ^~~~~~~~~~
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:243:9: note:   template argument deduction/substitution failed:
/workspaces/bolt/bolt/functions/lib/SIMDComparisonUtil.h:75:74: note:   mismatched types ‘const Tp*’ and ‘xsimd::batch<signed char>’
   75 |         xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 48)));
      |                                                                          ^
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:240:9: note: candidate: ‘template<class ... Ts> xsimd::batch_bool<T, A>::batch_bool(bool, bool, Ts ...) [with Ts = {Ts ...}; T = signed char; A = xsimd::neon64]’
  240 |         batch_bool(bool val0, bool val1, Ts... vals) noexcept;
      |         ^~~~~~~~~~
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:240:9: note:   template argument deduction/substitution failed:
/workspaces/bolt/bolt/functions/lib/SIMDComparisonUtil.h:75:74: note:   candidate expects at least 2 arguments, 1 provided
   75 |         xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 48)));
      |                                                                          ^
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:866:12: note: candidate: ‘xsimd::batch_bool<T, A>::batch_bool(register_type) [with T = signed char; A = xsimd::neon64; register_type = std::conditional<false, __Int8x16_t, __Uint8x16_t>::type]’
  866 |     inline batch_bool<T, A>::batch_bool(register_type reg) noexcept
      |            ^~~~~~~~~~~~~~~~
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:866:55: note:   no known conversion for argument 1 from ‘xsimd::batch<signed char>’ to ‘xsimd::batch_bool<signed char, xsimd::neon64>::register_type’ {aka ‘std::conditional<false, __Int8x16_t, __Uint8x16_t>::type’}
  866 |     inline batch_bool<T, A>::batch_bool(register_type reg) noexcept
      |                                         ~~~~~~~~~~~~~~^~~
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:1006:12: note: candidate: ‘xsimd::batch_bool<T, A>::batch_bool(bool) [with T = signed char; A = xsimd::neon64]’
 1006 |     inline batch_bool<T, A>::batch_bool(bool val) noexcept
      |            ^~~~~~~~~~~~~~~~
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:1006:46: note:   no known conversion for argument 1 from ‘xsimd::batch<signed char>’ to ‘bool’
 1006 |     inline batch_bool<T, A>::batch_bool(bool val) noexcept
      |                                         ~~~~~^~~
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:236:9: note: candidate: ‘xsimd::batch_bool<T, A>::batch_bool() [with T = signed char; A = xsimd::neon64]’
  236 |         batch_bool() = default; ///< Create a batch initialized with undefined values.
      |         ^~~~~~~~~~
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:236:9: note:   candidate expects 0 arguments, 1 provided
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:223:11: note: candidate: ‘constexpr xsimd::batch_bool<signed char, xsimd::neon64>::batch_bool(const xsimd::batch_bool<signed char, xsimd::neon64>&)’
  223 |     class batch_bool : public types::get_bool_simd_register_t<T, A>
      |           ^~~~~~~~~~
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:223:11: note:   no known conversion for argument 1 from ‘xsimd::batch<signed char>’ to ‘const xsimd::batch_bool<signed char, xsimd::neon64>&’
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:223:11: note: candidate: ‘constexpr xsimd::batch_bool<signed char, xsimd::neon64>::batch_bool(xsimd::batch_bool<signed char, xsimd::neon64>&&)’
/home/code/.conan2/p/xsimd3368e44453bd9/p/include/xsimd/types/xsimd_batch.hpp:223:11: note:   no known conversion for argument 1 from ‘xsimd::batch<signed char>’ to ‘xsimd::batch_bool<signed char, xsimd::neon64>&&’
[244/600] Building CXX object bolt/dwio/parquet/reader/CMakeFiles/bolt_dwio_native_parquet_reader.dir/ParquetColumnReader.cpp.o

```

### Reproduction Steps

Can reproduct inside devcontainer when build in Mac OS

<img width="4228" height="922" alt="Image" src="https://github.com/user-attachments/assets/6f205ed5-bab8-424a-8eba-40ac13f240ba" />

### Bolt Version / Commit ID

0de1e4a

### System Configuration

```markdown
- **OS**: (e.g. Ubuntu 22.04, CentOS 7)
- **Compiler**: (e.g. GCC 11, Clang 14)
- **Build Type**: (Debug / Release / RelWithDebInfo)
- **CPU Arch**: (e.g. x86_64 AVX2, ARM64)
- **Framework**: (e.g. Spark 3.3, PrestoDB)
```

### Logs / Stack Trace

```shell

```

### Expected Behavior

_No response_

### Additional context

_No response_

--- a/bolt/functions/lib/SIMDComparisonUtil.h
+++ b/bolt/functions/lib/SIMDComparisonUtil.h
@@ -56,23 +56,23 @@ inline uint64_t to64Bits(const int8_t* resultData) {
   uint64_t res = 0UL;
   if constexpr (numScalarElements == 64) {
     res = simd::toBitMask(
-        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData)));
+        (d_type::load_unaligned(resultData) != d_type(int8_t(0))));
   } else if constexpr (numScalarElements == 32) {
     auto* addr = reinterpret_cast<uint32_t*>(&res);
     *(addr) = simd::toBitMask(
-        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData)));
+        (d_type::load_unaligned(resultData) != d_type(int8_t(0))));
     *(addr + 1) = simd::toBitMask(
-        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 32)));
+        (d_type::load_unaligned(resultData + 32) != d_type(int8_t(0))));
   } else if constexpr (numScalarElements == 16) {
     auto* addr = reinterpret_cast<uint16_t*>(&res);
     *(addr) = simd::toBitMask(
-        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData)));
+        (d_type::load_unaligned(resultData) != d_type(int8_t(0))));
     *(addr + 1) = simd::toBitMask(
-        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 16)));
+        (d_type::load_unaligned(resultData + 16) != d_type(int8_t(0))));
     *(addr + 2) = simd::toBitMask(
-        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 32)));
+        (d_type::load_unaligned(resultData + 32) != d_type(int8_t(0))));
     *(addr + 3) = simd::toBitMask(
-        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 48)));
+        (d_type::load_unaligned(resultData + 48) != d_type(int8_t(0))));
   }
   return res;
 }

--- a/bolt/functions/sparksql/tests/ComparisonsTest.cpp
+++ b/bolt/functions/sparksql/tests/ComparisonsTest.cpp
@@ -29,10 +29,12 @@
  */
 
 #include "bolt/common/base/tests/GTestUtils.h"
+#include "bolt/functions/lib/SIMDComparisonUtil.h"
 #include "bolt/functions/sparksql/tests/SparkFunctionBaseTest.h"
 #include "bolt/vector/DecodedVector.h"
 
 #include <bolt/vector/SimpleVector.h>
+#include <array>
 namespace bytedance::bolt::functions::sparksql::test {
 namespace {
 
@@ -702,6 +704,34 @@ TEST_F(ComparisonsTest, testSIMDComparsion) {
   runSIMDCompareAndAssert<TypeKind::TIMESTAMP>(1024);
   runSIMDCompareAndAssert<TypeKind::VARCHAR>(686);
   runSIMDCompareAndAssert<TypeKind::VARBINARY>(777);
+}
+
+TEST_F(ComparisonsTest, to64BitsTreatsNonZeroAsTrue) {
+  std::array<int8_t, 64> resultData{};
+  resultData.fill(0);
+
+  resultData[0] = 1;
+  resultData[1] = -1;
+  resultData[7] = 2;
+  resultData[8] = -3;
+  resultData[15] = 4;
+  resultData[16] = -5;
+  resultData[23] = 6;
+  resultData[31] = -7;
+  resultData[32] = 8;
+  resultData[47] = -9;
+  resultData[63] = 10;
+
+  uint64_t expectedMask = 0;
+  for (size_t i = 0; i < resultData.size(); ++i) {
+    if (resultData[i] != 0) {
+      expectedMask |= (1ULL << i);
+    }
+  }
+
+  EXPECT_EQ(
+      bytedance::bolt::functions::detail::to64Bits(resultData.data()),
+      expectedMask);
 }
 
 TEST_F(ComparisonsTest, lessthan) {

